### PR TITLE
planner: handle single partition in IndexJoin correctly

### PR DESCRIPTION
**Pull request by eurekaka**
### What problem does this PR solve? 

Fix https://github.com/pingcap/tidb/issues/12549

### What is changed and how it works?

Set `isPartition` and `physicalTableID` for `PhysicalIndexScan` correctly when the inner child is a single partition of a partitioned table.

### Check List 

Tests 

 - Unit test

Code changes

N/A

Side effects

N/A

Related changes

 - Need to cherry-pick to the release branch

Release note

N/A

### CODE REVIEW

File: planner/core/exhaust_physical_plans.go [hunk 1]
```diff
@@ -704,6 +704,8 @@ func (p *LogicalJoin) constructInnerIndexScanTask(
 		KeepOrder:        keepOrder,
 		Ranges:           ranger.FullRange(),
 		rangeInfo:        rangeInfo,
+		isPartition:      ds.isPartition,
```
Comment (lamxTyler) [inline planner/core/exhaust_physical_plans.go]: Seems that it is also not set in `(b *PlanBuilder) buildPhysicalIndexLookUpReader`, should we fix it in this pr?

Comment (eurekaka) [reply planner/core/exhaust_physical_plans.go]: It looks like `admin check table` and `admin check index` does not work for partitioned table at all? I didn't find code converting partitioned table to `union all` for these 2 command, @crazycs520 PTAL

**Comment by zz-jason** [comment]
@eurekaka please resolve the conflicts.

File: planner/core/exhaust_physical_plans.go (hunk 1 above)
Comment (zimulala) [reply planner/core/exhaust_physical_plans.go]: Yes. `admin check table` and `admin check index` call `buildPhysicalIndexLookUpReader` directly. I think they don't set `isPartition`

**Pull request merged**